tests/test_packages/test_connections/test_p2p_libp2p/test_fault_tolerance.py

```diff
@@ -71,174 +65,119 @@ def change_state_and_wait(
-
-        cls.connection2 = _make_libp2p_connection(relay=False, entry_peers=[relay_peer])
-        cls.multiplexer2 = Multiplexer([cls.connection2], protocols=[DefaultMessage])
-        cls.multiplexer2.connect()
-        cls.log_files.append(cls.connection2.node.log_file)
-        cls.multiplexers.append(cls.multiplexer2)
+class TestLibp2pConnectionRelayNodeRestart(BaseTestLibp2pRelay):
+    """Test that connection will reliably forward envelopes after its relay node restarted"""
+
+    def setup(self):
+        """Set up the individual test method"""
+
+        self.genesis = self.make_connection()
+        genesis_peer = self.genesis.node.multiaddrs[0]
+        self.relay = self.make_connection(entry_peers=[genesis_peer])
+        relay_peer = self.relay.node.multiaddrs[0]
+        self.connection1 = self.make_connection(relay=False, entry_peers=[relay_peer])
+        self.connection2 = self.make_connection(relay=False, entry_peers=[relay_peer])
+
+        # create references for disconnecting and reconnecting
+        self.multiplexer_genesis = self.multiplexers[0]
+        self.multiplexer_relay = self.multiplexers[1]
+        self.multiplexer1 = self.multiplexers[2]
+        self.multiplexer2 = self.multiplexers[3]
+
+    def teardown(self):
+        """Teardown"""
+        self.disconnect()
+        self.multiplexers.clear()
 
     def test_connection_is_established(self):
         """Test connection established."""
-        assert self.relay.is_connected is True
-        assert self.connection.is_connected is True
-        assert self.connection2.is_connected is True
+        assert self.all_multiplexer_connections_connected
 
     def test_envelope_routed_from_peer_after_relay_restart(self):
         """Test envelope routed from third peer after relay restart."""
 
+        to = self.connection1.address
         sender = self.genesis.address
-        to = self.connection.address
 
-        envelope = self.enveloped_default_message(to=to, sender=sender)
-        self.multiplexer_genesis.put(envelope)
-        delivered_envelope = self.multiplexer.get(block=True, timeout=TIMEOUT)
+        def attempt_sending():
+            envelope = self.enveloped_default_message(to=to, sender=sender)
+            self.multiplexer_genesis.put(envelope)
+            delivered_envelope = self.multiplexer1.get(block=True, timeout=TIMEOUT)
+            assert delivered_envelope is not None
+            assert self.sent_is_delivered_envelope(envelope, delivered_envelope)
 
-        assert delivered_envelope is not None
-        assert self.sent_is_delivered_envelope(envelope, delivered_envelope)
+        assert self.multiplexer_relay.is_connected
+        attempt_sending()
 
         self.multiplexer_relay.disconnect()
         self.change_state_and_wait(self.multiplexer_relay, expected_is_connected=False)
+        assert not self.multiplexer_relay.is_connected
+
+        with pytest.raises(Empty):
+            attempt_sending()
```

Comment:
added checks for `is_connected` and `not is_connected`. Also added the pytest raise to verify that an error is indeed raised when disconnected, validating the presumed functional effect of the disconnect